<?xml version="1.0" encoding="ISO-8859-15"?>
<launch>

  <arg name="play_bag" default="true" />
  <!-- <arg name="bag_file" default="/Users/helen/data/euroc_datasets/V2_01_easy.bag"/> -->
  <arg name="bag_file" default="/Users/helen/data/euroc_datasets/euroc_v1_01_easy.bag"/>

  <arg name="voxel_size" default="0.05"/>
  <arg name="generate_esdf" default="false" />

  <node name="player" pkg="rosbag" type="play" output="screen" args=" -r 1.0 --clock $(arg bag_file)" if="$(arg play_bag)"/>


   <!-- Run stereo_dense_reconstruction_node-->
  <node name="stereo_dense_reconstruction_node" pkg="stereo_dense_reconstruction_nodes" type="stereo_dense_reconstruction_node" clear_params="true" output="screen">
    <remap from="image_cam0"          to="cam0/image_raw" />
    <remap from="image_cam1"          to="cam1/image_raw" />
    <remap from="calib_cam0"          to="cam0/calibration" />
    <remap from="calib_cam1"          to="cam1/calibration" />

    <!-- Specify the tf-name of the stereo camera's reference frame. This is not mandatory if optimizer is turned off.-->
    <remap from="/ref"                to="/imu0" />

    <!-- Specify if images need to be undistorted -->
    <param name="undistort_images" value="true"/>

    <param name="calibration_from_topic" value="false"/>

     <!-- Optional: Specify which optional images/pcls should be published
     <param name="publish_disparity_map" value="true"/>
     <param name="publish_pcl_of_vision" value="true"/>
     <param name="publish_fused_depthmap" value="true"/>
     <param name="publish_fused_image" value="true"/>-->

    <!-- Optional: Specify if default transform world->cam should be published if none has been found -->
    <param name="publish_world_cam_default_tf" value="true"/>

    <!-- Load parameters -->
    <rosparam file="$(find stereo_dense_reconstruction_nodes)/cfg/ParamsHalfRes.yaml"/>

    <!-- Load stereo camera calibration from yaml file (only used if visensor_msgs not found)-->
    <rosparam file="$(find mav_startup)/parameters/mavs/ijrr_euroc_datasets/camchain.yaml"/>
  </node>

  <node name="voxblox_node" pkg="voxblox_ros" type="tsdf_server" output="screen" args="-alsologtostderr" clear_params="true">
    <remap from="pointcloud" to="stereo_dense_reconstruction/pcl"/>
    <param name="tsdf_voxel_size" value="$(arg voxel_size)" />
    <param name="tsdf_voxels_per_side" value="16" />
    <param name="voxel_carving_enabled" value="true" />
    <param name="color_mode" value="colors" />
    <param name="use_tf_transforms" value="false" />
    <param name="verbose" value="true" />
    <!--
    <remap from="transform" to="vicon/firefly_sbx/firefly_sbx" /> -->
    <remap from="transform" to="/groundtruth_transform" />
    <param name="update_mesh_every_n_sec" value="1.0" />
    <param name="generate_esdf" value="$(arg generate_esdf)" />
    <param name="slice_level" value="1.0" />
    <param name="method" value="fast" />
    <param name="anti_grazing" value="false" />
    <param name="use_const_weight" value="false" />

    <rosparam file="$(find voxblox_ros)/cfg/euroc_dataset.yaml"/>

    <param name="mesh_filename" value="$(find voxblox_ros)/mesh_results/$(anon euroc).ply" />
  </node>
</launch>
